cookiecutter/repository.py

```diff
@@ -3,13 +3,15 @@
 import re
 
 from cookiecutter.exceptions import RepositoryNotFound
-from cookiecutter.vcs import clone
+from cookiecutter.vcs import clone, REPO_TYPES
 from cookiecutter.zipfile import unzip
 
 REPO_REGEX = re.compile(
     r"""
 # something like git:// ssh:// file:// etc.
-((((git|hg)\+)?(git|ssh|file|https?):(//)?)
+(((("""
+    + '|'.join(REPO_TYPES.keys())
+    + r""")\+)?(git|svn|ssh|file|https?):(//)?)
```

Comment:
I would say to just write:
```python
REPO_REGEX = re.compile(
    r"((((git|hg|svn)\+)?(git|svn|ssh|file|https?):(//)?)"  # git:// git+https:// ...
    r"|(\w+@[\w\.]+))",  # or something like user@...
    re.VERBOSE,
)
```
It's simpler and more readable. @Theta-Dev what do you think?

(we should rework this regex in the future anyway, as it is right now it's overly permissive and allows no sense patterns like `git+svn://` etc)